libraries/ESP8266WiFiMesh/src/ESP8266WiFiMesh.cpp

```diff
@@ -18,151 +18,582 @@
   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 */
 
-#include <Arduino.h>
 #include <ESP8266WiFi.h>
 #include <WiFiClient.h> 
 #include <WiFiServer.h>
 
 #include "ESP8266WiFiMesh.h"
 
-#define SSID_PREFIX      		"Mesh_Node"
-#define SERVER_IP_ADDR			"192.168.4.1"
-#define SERVER_PORT				4011
+#define SERVER_IP_ADDR      "192.168.4.1"
 
-ESP8266WiFiMesh::ESP8266WiFiMesh(uint32_t chip_id, std::function<String(String)> handler)
-: _server(SERVER_PORT)
+const IPAddress ESP8266WiFiMesh::empty_IP = IPAddress();
+const uint32_t ESP8266WiFiMesh::lwip_version_203_signature[3] {2,0,3};
+
+String ESP8266WiFiMesh::last_ssid = "";
+bool ESP8266WiFiMesh::static_IP_activated = false;
+
+// IP needs to be at the same subnet as server gateway (192.168.4 in this case). Station gateway ip must match ip for server.
+IPAddress ESP8266WiFiMesh::static_IP = empty_IP;
+IPAddress ESP8266WiFiMesh::gateway = IPAddress(192,168,4,1);
+IPAddress ESP8266WiFiMesh::subnet_mask = IPAddress(255,255,255,0);
+
+std::vector<NetworkInfo> ESP8266WiFiMesh::connection_queue = {};
+std::vector<TransmissionResult> ESP8266WiFiMesh::latest_transmission_outcomes = {};
+
+ESP8266WiFiMesh::ESP8266WiFiMesh(std::function<String(String, ESP8266WiFiMesh *)> requestHandler, std::function<transmission_status_t(String, ESP8266WiFiMesh *)> responseHandler, 
+                                 std::function<void(int, ESP8266WiFiMesh *)> networkFilter, String mesh_password, String mesh_name, String node_id, bool verbose_mode, uint8 mesh_wifi_channel, int server_port) 
+                                 : _server(server_port), _lwip_version{0, 0, 0}
+{
+  updateNetworkNames(mesh_name, (node_id != "" ? node_id : Uint64ToString(ESP.getChipId())));
+  _requestHandler = requestHandler;
+  _responseHandler = responseHandler;
+  _mesh_wifi_channel = mesh_wifi_channel;
+  _server_port = server_port;
+  _mesh_password = mesh_password;
+  _verbose_mode = verbose_mode;
+  _networkFilter = networkFilter;
+  storeLwipVersion();
+}
+
+void ESP8266WiFiMesh::verboseModePrint(String string_to_print, bool newline)
+{
+  if(_verbose_mode)
+  {
+    if(newline)
+      Serial.println(string_to_print);
+    else
+      Serial.print(string_to_print);
+  }
+}
+
+/**
+ * Note that using a base higher than 16 increases likelihood of randomly generating ssid strings containing controversial words. 
+ * 
+ * @param number The number to convert to a string with radix "base".
+ * @param base The radix to convert "number" into. Must be between 2 and 36.
+ * @returns A string of "number" encoded in radix "base".
+ */
+String ESP8266WiFiMesh::Uint64ToString(uint64_t number, byte base)
+{
+  String result = "";
+
+  while(number > 0)
+  {
+    result = String((uint32_t)(number % base), base) + result;
+    number /= base;
+  }
+
+  return (result == "" ? "0" : result);
+}
```

Comment:
Does it make sense to put this function in a standalone file?

Reply:
Sure, I could do that. Do you mean as a separate utility class or just definitions in a separate .cpp?

Reply:
Just movemto some separate file would be enough. The functions are generic and not specific to the contents of these files.